src/Sylius/Bundle/WebBundle/Behat/WebContext.php

```diff
@@ -703,6 +704,17 @@ public function iChangeMyPasswordTo($newPassword)
         $this->pressButton('Save changes');
     }
 
+    /**
+     * @Then Coupons code length for :promotionName should be :length
+     */
+    public function couponsCodeLengthForShouldBe($promotionName, $length)
+    {
+        $couponRepository = $this->getRepository('promotion_coupon');
```

Comment:
You can use `$entityManager->clear($entityName)` to detach that promotion from Entity Manager and fetch updated coupons by `$promotion->getCoupons()`.